src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Schemas/ProductType.cs

```diff
@@ -2,49 +2,187 @@
-            Field<ListGraphType<PropertyType>>("properties", resolve: context => context.Source.CatalogProduct.Properties);
 
-            Field<ListGraphType<PriceType>>("prices", arguments: new QueryArguments(new QueryArgument<StringGraphType> { Name = "currency", Description = "currency" }), resolve: context =>
+            Field<StringGraphType>(
+                "brandName",
+                description: "Get brandName for product.",
+                resolve: context =>
+                {
+                    var brandName = context.Source.CatalogProduct.Properties
+                        ?.FirstOrDefault(x => x.Name == "Brand")
+                        ?.Values
+                        ?.FirstOrDefault(x => x.Value != null)
+                        ?.Value;
+
+                    return brandName;
+                });
+
+            FieldAsync<VariationType>("masterVariation", resolve: async context =>
             {
-                var result = (context.Source).Prices;
-                var currency = context.GetArgument<string>("currency");
-                if (currency != null)
+                if (context.Source.CatalogProduct.MainProductId == null)
                 {
-                    result = result.Where(x => x.Currency.EqualsInvariant(currency)).ToList();
+                    var product = context.Source.CatalogProduct;
+
+                    return new ExpVariation
+                    {
+                        Product = product,
+                        Prices = context.Source.ProductPrices
+                    };
                 }
-                return result;
+
+                var response = await mediator.Send(new LoadProductQuery
+                {
+                    Ids = new[] { context.Source.CatalogProduct.MainProductId },
+                    IncludeFields = context.SubFields.Values.GetAllNodesPaths()
+                });
+
+                return response.Products.Select(expProduct => new ExpVariation
+                {
+                    Product = expProduct.CatalogProduct,
+                    Prices = expProduct.ProductPrices
+                }).FirstOrDefault();
             });
 
+            FieldAsync<ListGraphType<VariationType>>("variations", resolve: async context =>
+            {
+                var variationsIds = context.Source.VariationIds;
+                var response = await mediator.Send(new LoadProductQuery
+                {
+                    Ids = variationsIds.ToArray(),
+                    IncludeFields = context.SubFields.Values.GetAllNodesPaths()
+                });
+
+                return response.Products.Select(expProduct => new ExpVariation
+                {
+                    Product = expProduct.CatalogProduct,
+                    Prices = expProduct.ProductPrices
+                });
+            });
+
+            //Field<ListGraphType<PropertyType>>("properties", resolve: context => context.Source.CatalogProduct.Properties);
```

Comment:
Why this commented out? There must be returned flat expended (you choose strange name Pivot but I not sure that this a good name for this) properties that have type `Product`.